### PR TITLE
Make `Build` and `Test` checks triggered

### PR DESCRIPTION
**Pull request by ArthurBandaryk**
For now this checks will be triggered on any push on any branch.

### CODE REVIEW

**Comment by while-false** [comment]
Hi Arthur, I am happy to see this in there. There seem to be an unexpected side effect: we are running every action twice. It looks like it is triggered on push AND then again on pull.
Can I ask you to find out if there is a way to make this an OR so we only run one of them? If there is no way of doing this, we might want to think about whether we want to have `build` and `test` run on every push.
That is: if one can not easily make it only run the `build` and `test` actions, we might want to remove the 'push' trigger from the yaml for these checks. For the formatting checks that are quick, I think we should keep it, even if redundant.

**Comment by ArthurBandaryk** [comment]
> Hi Arthur, I am happy to see this in there. There seem to be an unexpected side effect: we are running every action twice. It looks like it is triggered on push AND then again on pull.
> Can I ask you to find out if there is a way to make this an OR so we only run one of them? If there is no way of doing this, we might want to think about whether we want to have `build` and `test` run on every push.
> That is: if one can not easily make it only run the `build` and `test` actions, we might want to remove the 'push' trigger from the yaml for these checks. For the formatting checks that are quick, I think we should keep it, even if redundant.

Yaaaaaay, Gorm:) After a couple of hours I finally found the solution!!! A simple `if` statement will help us!
```
if: github.event_name == 'push' || github.event.pull_request.head.repo.full_name != '3rdparty/stout-grpc'

```

This means that we want to run on external PRs, but not on our own internal PRs as they'll be run by the push to the branch.

File: .github/workflows/macos.yml [hunk 1]
```diff
@@ -3,14 +3,17 @@ name: macOS
...
   pull_request:
-    branches: [master]
+    branches:
+      - "master"
 
 jobs:
   build-and-test:
     name: Build and Test
     runs-on: macos-latest
+    if: github.event_name == 'push' || github.event.pull_request.head.repo.full_name != '3rdparty/stout-grpc'
```
Comment (rjhuijsman) [inline .github/workflows/macos.yml:16]: Two things about this:
(1) Can we please add a comment explaining why this is here?
(2) Does this line make lines 9 and 10 (triggering on pull request to master) obsolete?

Comment (ArthurBandaryk) [reply .github/workflows/macos.yml:16]: 1) Of course we can:)
2) Definitely not cause we wanna run this checks on external PRs:)

Comment (rjhuijsman) [reply .github/workflows/macos.yml:16]: Re 2: good point!

**Review by while-false — APPROVED** [review]
LGTM, happy to see all the changes in there! Thanks Arthur.

**Pull request merged**